compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -594,41 +510,169 @@ public STNode resumeParsing(ParserRuleContext context, Object... args) {
                 return parseMatchKeyword();
+            case PARAMETERIZED_TYPE:
+                return parseParameterizedTypeKeyword();
+            case LOCK_KEYWORD:
+                return parseLockKeyword();
+            case STRING_KEYWORD:
+                return parseStringKeyword();
+            case TRANSACTIONAL_KEYWORD:
+                return parseTransactionalKeyword();
+            default:
+                return resumeOtherNodesParsing(context, args);
+        }
+    }
+
+    public STNode resumeActionOrExpressionNodesParsing(ParserRuleContext context, Object... args) {
+        switch (context) {
+            case TERMINAL_EXPRESSION:
+                return parseTerminalExpression((STNode) args[0], (boolean) args[1], (boolean) args[2],
+                        (boolean) args[3]);
+            case EXPRESSION_RHS:
+                return parseExpressionRhs((OperatorPrecedence) args[0], (STNode) args[1], (boolean) args[2],
+                        (boolean) args[3], (boolean) args[4], (boolean) args[5]);
+            case CONSTANT_EXPRESSION_START:
+                return parseSimpleConstExprInternal();
+            case BINDING_PATTERN_OR_EXPR_RHS:// 2 resume-points : parseTypeDescOrExprRhs()
+                return parseTypedBindingPatternOrExprRhs((STNode) args[0], (boolean) args[1]);
+            case TYPE_DESC_OR_EXPR_RHS:
+                return parseTypeDescOrExprRhs((STNode) args[0]);
+            case WAIT_FUTURE_EXPR_END:
+                return parseWaitFutureExprEnd((int) args[0]);
+            case MEMBER_ACCESS_KEY_EXPR_END:
+                return parseMemberAccessKeyExprEnd();
+            case BRACED_EXPR_OR_ANON_FUNC_PARAM_RHS:
+                return parseBracedExprOrAnonFuncParamRhs((STNode) args[0], (STNode) args[1], (boolean) args[2]);
+            case EXPR_FUNC_BODY_START:
+                return parseDoubleRightArrow();
+            case REMOTE_CALL_OR_ASYNC_SEND_RHS:
+                return parseRemoteCallOrAsyncSendActionRhs((STNode) args[0], (boolean) args[1], (STNode) args[2]);
+            default:
+                return resumeKeywordParsing(context, args);
+        }
+    }
+
+    public STNode resumeStatementNodeParsing(ParserRuleContext context, Object... args) {
+        switch (context) {
+            case STATEMENT:
+                return parseStatement();
+            case STATEMENT_WITHOUT_ANNOTS:
+                return parseStatement((STNode) args[0]);
+            case STMT_START_WITH_EXPR_RHS:
+                return parseStatementStartWithExprRhs((STNode) args[0]);
+            case STMT_START_BRACKETED_LIST_MEMBER:
+                return parseStatementStartBracketedListMember();
+            case VAR_DECL_STMT_RHS:
+                return parseVarDeclRhs((STNode) args[0], (STNode) args[1], (STNode) args[2], (boolean) args[3]);
+            case COMPOUND_ASSIGNMENT_STMT:// following method is only referred in resume parsing
+                return parseCompoundAssignmentStmt();
```

Comment:
`parseCompoundAssignmentStmt()` is only called within `resumeParsing()`
No other usage. Do we need that method?

Reply:
I don't think we do. I ll remove it.

Reply:
when removing can remove those enclosing `COMPOUND_ASSIGNMENT_STMT` ctx as well. Seems it is referred in error handler also.
Same case with `parseNilTypeDescriptor()` and its `NIL_TYPE_DESCRIPTOR` ctx. 

I think we may have to recheck the usage of those ctxs in error handler as well @SupunS WDYT?

Reply:
Yeah if the method doesn't go to error recovery (eg: parseNilTypeDescriptor method), then we don't need them in the resume method also

Reply:
@SupunS yea. In my second comment what I meant was , if we are removing those two methods we are eliminating
`startContext(ParserRuleContext.COMPOUND_ASSIGNMENT_STMT);` and `startContext(ParserRuleContext.NIL_TYPE_DESCRIPTOR);` calls from BallerinaParser. 

But in the error handler still we are starting new ctxs for those ctxs. So I see some kind of inconsistency. 
May be we can create an issue to check on that and remove those unnecessary ctx starts from error handler?